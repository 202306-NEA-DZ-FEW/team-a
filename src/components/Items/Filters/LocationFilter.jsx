import { Menu, Transition } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { HiOutlineLocationMarker } from "react-icons/hi";
import { IoMdArrowDropdown } from "react-icons/io";

import getAllStates from "@/lib/getAllStates";

function LocationFilter({ t, queryParams }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  //This is to test the LocationFilter
  const states = getAllStates(t);

  return (
    <>
      <Menu as='div' className='relative w-full flex justify-center text-left'>
        <Menu.Button className='flex items-center gap-2 w-full justify-between bg-base-300 input-sm rounded-full tracking-wider'>
          <div className='flex gap-2 items-center'>
            <HiOutlineLocationMarker className='w-5 h-5' />
            <span>
              {queryParams.location
                ? t(`states:${queryParams.location}`)
                : t("itemsPage:location")}
            </span>
          </div>
          <IoMdArrowDropdown />
        </Menu.Button>

        <Transition
          as={Fragment}
          enter='transition ease-out duration-100'
          enterFrom='transform opacity-0 scale-95'
          enterTo='transform opacity-100 scale-100'
          leave='transition ease-in duration-75'
          leaveFrom='transform opacity-100 scale-100'
          leaveTo='transform opacity-0 scale-95'
        >
          <Menu.Items className='absolute max-h-52 no-scrollbar p-2 overflow-y-scroll bg-base-100 z-10 rounded-lg top-10 w-full ring-1 ring-black ring-opacity-5 focus:outline-none drop-shadow-xl'>
            {states.map(({ name, dataKey }) => (
              <Link
                scroll={false}
                key={dataKey}
                href={{
                  pathname: "/items",
                  query: { ...queryParams, location: dataKey },
                }}
              >
                <Menu.Item>
                  {({ active }) => (
                    <p
                      className={classNames(
                        active ? "bg-base-300" : "text-gray-500",
                        "block px-4 py-2 tracking-wider my-2 rounded-md"
                      )}
                    >
                      {name}
                    </p>
                  )}
                </Menu.Item>
              </Link>
            ))}
          </Menu.Items>
        </Transition>
      </Menu>
    </>
  );
}

export default LocationFilter;
