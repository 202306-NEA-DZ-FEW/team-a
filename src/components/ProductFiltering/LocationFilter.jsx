import { Menu, Transition } from "@headlessui/react";
import Link from "next/link";
import { Fragment } from "react";
import { HiOutlineLocationMarker } from "react-icons/hi";
import { IoMdArrowDropdown } from "react-icons/io";

import getAllStates from "@/lib/getAllStates";

function LocationFilter({ t, queryParams }) {
  function classNames(...classes) {
    return classes.filter(Boolean).join(" ");
  }
  //This is to test the LocationFilter
  const states = getAllStates(t);

  return (
    <>
      <Menu
        as='div'
        className='relative w-full md:w-52 flex justify-center text-left'
      >
        <Menu.Button className='flex items-center gap-2 w-full justify-between bg-white px-3 py-2 text-sm font-semibold btn-sm rounded-xl text-gray-900 hover:bg-gray-50 btn-primary'>
          <div className='flex gap-2 items-center'>
            <HiOutlineLocationMarker className='w-5 h-5' />
            <span>{t("productsPage:location")}</span>
          </div>
          <IoMdArrowDropdown />
        </Menu.Button>

        <Transition
          as={Fragment}
          enter='transition ease-out duration-100'
          enterFrom='transform opacity-0 scale-95'
          enterTo='transform opacity-100 scale-100'
          leave='transition ease-in duration-75'
          leaveFrom='transform opacity-100 scale-100'
          leaveTo='transform opacity-0 scale-95'
        >
          <Menu.Items className='absolute max-h-52 overflow-y-scroll bg-white z-10 rounded-lg top-10 w-full md:w-52'>
            <div className='py-1'>
              {states.map(({ name, dataKey }) => (
                <Link
                  scroll={false}
                  key={dataKey}
                  href={{
                    pathname: "/products",
                    query: { ...queryParams, location: dataKey },
                  }}
                >
                  <Menu.Item>
                    {({ active }) => (
                      <p
                        className={classNames(
                          active
                            ? "bg-gray-100 text-gray-900"
                            : "text-gray-700",
                          "block px-4 py-2 text-sm my-2"
                        )}
                      >
                        {name}
                      </p>
                    )}
                  </Menu.Item>
                </Link>
              ))}
            </div>
          </Menu.Items>
        </Transition>
      </Menu>
    </>
  );
}

export default LocationFilter;
