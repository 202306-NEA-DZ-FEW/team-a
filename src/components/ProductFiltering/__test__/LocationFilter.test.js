import renderer from "react-test-renderer";

import LocationFilter from "../LocationFilter";

it("renders correctly", () => {
  const queryParams = {};
  const mockT = jest.fn();
  const tree = renderer
    .create(<LocationFilter queryParams={queryParams} t={mockT} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
