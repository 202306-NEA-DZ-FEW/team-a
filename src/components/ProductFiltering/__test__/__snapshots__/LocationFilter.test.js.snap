// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="relative w-full md:w-52 flex justify-center text-left"
  data-headlessui-state=""
>
  <button
    aria-expanded={false}
    aria-haspopup="menu"
    className="flex items-center gap-2 w-full justify-between bg-white px-3 py-2 text-sm font-semibold btn-sm rounded-xl text-gray-900 hover:bg-gray-50 btn-primary"
    data-headlessui-state=""
    id="headlessui-menu-button-:r0:"
    onClick={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    type="button"
  >
    <div
      className="flex gap-2 items-center"
    >
      <svg
        aria-hidden="true"
        className="w-5 h-5"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="2"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M17.657 16.657L13.414 20.9a1.998 1.998 0 01-2.827 0l-4.244-4.243a8 8 0 1111.314 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <path
          d="M15 11a3 3 0 11-6 0 3 3 0 016 0z"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </svg>
      <span />
    </div>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M128 192l128 128 128-128z"
      />
    </svg>
  </button>
</div>
`;
